src/WriterModuleConfig/Field.h

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#pragma once
+
+#include "logger.h"
+#include <nlohmann/json.hpp>
+#include <string>
+
+namespace WriterModule {
+class Base;
+}
+
+namespace WriterModuleConfig {
+
+using namespace nlohmann;
+
+class FieldBase {
+public:
+  FieldBase(WriterModule::Base *Ptr, std::vector<std::string> const &Keys);
+  FieldBase(WriterModule::Base *Ptr, std::string const &Key)
+      : FieldBase(Ptr, std::vector<std::string>{Key}) {}
+  virtual ~FieldBase() {}
+  virtual void setValue(std::string const &NewValue) = 0;
+  bool hasDefaultValue() const { return GotDefault; }
+  auto getKeys() const { return FieldKeys; }
+  bool isRequried() const { return FieldRequired; }
+
+protected:
+  bool GotDefault{true};
+  void makeRequired();
+
+private:
+  std::vector<std::string> FieldKeys;
+  bool FieldRequired{false};
+};
+
+template <class FieldType> class Field : public FieldBase {
+public:
+  Field(WriterModule::Base *WriterPtr, std::string const &Key,
+        FieldType DefaultValue)
+      : FieldBase(WriterPtr, Key), FieldValue(DefaultValue) {}
+  Field(WriterModule::Base *WriterPtr, std::vector<std::string> Keys,
+        FieldType DefaultValue)
+      : FieldBase(WriterPtr, Keys), FieldValue(DefaultValue) {}
+
+  void setValue(std::string const &ValueString) override {
+    setValueImpl<FieldType>(ValueString);
+  }
+
+  FieldType getValue() { return FieldValue; };
+
+  operator FieldType() const { return FieldValue; }
```

Comment:
```suggestion
  explicit operator FieldType() const { return FieldValue; }
```

Reply:
In this particular case: no. Fixing all the implicit conversions (that has been there from the beginning) is a part of another PR, not this one.

Reply:
ok